### PR TITLE
Login func

### LINKED ISSUE #12: Make login functionalities for google authentication
You can follow this tutorial.
https://www.youtube.com/watch?v=S_sV6bYWKXQ

Build functionalities to login with google auth. 
When authentication is successful, callback should be in this route "/login-validation"
So, make a login-validation route where basic validation will happen and user's information will be stored into the database.

After that user will be logged in and be moved to "/home" page. In the home page you can just console log user has logged in.

### PR DESCRIPTION
**Pull request by AR-Aourangazeb-Alif**
## Description

Added Login Functionalities according to the #12 issue

closes #12 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have run `npm run format`
- [x] I have run `npm run lint`
- [x] I have updated changelog
- [ ] My PR requires .env changes
- [x] My PR requires `npm install`
- [x] My PR requires additional attention from Arafat

### CODE REVIEW

File: app/login/page.js [hunk 1]
```diff
@@ -3,15 +3,21 @@ import Image from 'next/image'
...
-import { useRouter } from 'next/navigation'
+import { useState } from 'react'
 
 const LoginPage = () => {
-  const { handleGoogleLogin } = useAuth()
-  const router = useRouter()
+  const { handleGoogleLogin, showLoginError, setShowLoginError } = useAuth()
+
+  // Handle Google auth
+  const handleGoogleSignin = () => {
```
Comment (Arafatmollik1) [inline app/login/page.js:12]: Why is handleGoogleLogin() inside the constant handleGoogleSignin? This variable is not being used.

File: app/login/page.js [hunk 2]
```diff
@@ -24,18 +30,26 @@ const LoginPage = () => {
...
+        <div className='relative w-full'>
+          <h1
+            className={`absolute -top-8 w-full rounded-[4px] p-1 text-center text-sm font-medium capitalize text-red-700 ${showLoginError ? 'block' : 'hidden'}`}
+          >
+            Failed to log in !
+          </h1>
+
+          <button
+            className='w-full rounded-[5px] border border-[#97BFD7] p-3 text-sm font-medium text-[#364C6F] transition-all hover:shadow-sm hover:shadow-[#97bed781]'
+            onClick={handleGoogleLogin}
```
Comment (Arafatmollik1) [inline app/login/page.js:48]: Did you meant to use handleGoogleSignin instead of handleGoogleLogin?

**Comment by Arafatmollik1** [comment]
@AR-Aourangazeb-Alif 
Hi, did you understand everything you wrote in [AuthContext.js](https://github.com/Arafatmollik1/fundy-tickets-nextjs/pull/15/files#diff-52a6a8ee944c88791f1ab1b81f1530589bad2036e31d516d123a4d7766c0c6bb)

Let me know If there is anything you don't understand and would like me to explain.

**Comment by Arafatmollik1** [comment]
@AR-Aourangazeb-Alif 

Your PR required me to do npm install. so please check the "My PR requires npm install" in this type situation.

Also, you marked My PR requires additional attention from Arafat. But did not write what.

In this case, you could write a comment in the PR that "Your need to download firebase SDK from  xx and then rename it to credentials.json and locate it to root directory"

Its okay for now.

File: app/login/page.js (hunk 2 above)
Comment (AR-Aourangazeb-Alif) [reply app/login/page.js:48]: Yes, this is a mistake from my side. I was suppose to use handleGoogleSignin instead of handleGoogleLogin.

**Comment by AR-Aourangazeb-Alif** [comment]
> @AR-Aourangazeb-Alif Hi, did you understand everything you wrote in [AuthContext.js](https://github.com/Arafatmollik1/fundy-tickets-nextjs/pull/15/files#diff-52a6a8ee944c88791f1ab1b81f1530589bad2036e31d516d123a4d7766c0c6bb)
> 
> Let me know If there is anything you don't understand and would like me to explain.

**Pull request closed without merging**